cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -198,7 +199,7 @@ runTransactionBuildCmd
           <$> readTxGovernanceActions eon proposalFiles
 
     -- the same collateral input can be used for several plutus scripts
-    let filteredTxinsc = toList $ Set.fromList txinsc
+    let filteredTxinsc = toList @(Set _) $ fromList txinsc
```

Comment:
Same here, I prefer keeping `Set.fromList`. It's more KISS.

Reply:
Good catch. I've replaced it with `nubOrd` which has the same asymptotics for unsorted lists.